src/main/java/org/openmaptiles/layers/Transportation.java

```diff
@@ -245,6 +255,40 @@ private static boolean isResidentialOrUnclassified(String highway) {
     return "residential".equals(highway) || "unclassified".equals(highway);
   }
 
+  private static boolean isTrunkForZ5(String highway, List<RouteRelation> routeRelations) {
+    // Allow trunk roads that are part of a nation's most important route network to show at z5
+    if (!"trunk".equals(highway)) {
+      return false;
+    }
+    return routeRelations.stream()
+      .map(RouteRelation::networkType)
+      .filter(Objects::nonNull)
+      .anyMatch(Z5_TRUNK_BY_NETWORK::contains);
+  }
+
+  private static boolean isMotorwayWithNetworkForZ4(List<RouteRelation> routeRelations) {
+    // All roads in network included in osm_national_network except gb-trunk and us-highway
+    return routeRelations.stream()
+      .map(RouteRelation::networkType)
+      .filter(Objects::nonNull)
+      .filter(nt -> !Z5_MOTORWAYS_BY_NETWORK.contains(nt))
+      .anyMatch(Z5_TRUNK_BY_NETWORK::contains);
+  }
+
+  private static boolean isMotorwayWoNetworkForZ4(List<RouteRelation> routeRelations) {
+    // All motorways without network (e.g. EU, Asia, South America)
+    return routeRelations.stream()
+      .map(RouteRelation::networkType)
+      .noneMatch(Objects::nonNull);
+  }
```

Comment:
I notice that this logic leads to short stretches of road in the US getting included at z4 because they have class=motorway without a route network and they are disconnected with the rest of the road because we exclude class=motorway/network=us-highway until z5 - is that the intent? 

I think it's because this logic is upgrading those trunks to motorways:

```java
          if (z5trunk && isMotorwayWithNetworkForZ4(routeRelations)) {
            clazz = FieldValues.CLASS_MOTORWAY;
```

<img width="721" alt="image" src="https://github.com/openmaptiles/planetiler-openmaptiles/assets/1480504/07455377-56e2-4a4a-b18c-5b63db2101fd">


https://protomaps.github.io/PMTiles/?url=https%3A%2F%2Fprotomaps.dev%2F%7Emsbarry%2Fomt-3.15-branch.pmtiles#map=4.8/36.05/-84.16